src/main/scala/xiangshan/mem/lsqueue/StoreQueue.scala

```diff
@@ -1459,7 +1459,7 @@ class StoreQueue(implicit p: Parameters) extends XSModule
   // misprediction recovery / exception redirect
   // invalidate sq term using robIdx
   for (i <- 0 until StoreQueueSize) {
-    needCancel(i) := (uop(i).robIdx.needFlush(io.brqRedirect) & !isVec(i) || isAfter(uop(i).robIdx, io.brqRedirect.bits.robIdx) && io.brqRedirect.valid && isVec(i)) && allocated(i) && !committed(i)
+    needCancel(i) := (uop(i).robIdx.needFlush(io.brqRedirect) && !vecExceptionFlag.valid || isAfter(uop(i).robIdx, io.brqRedirect.bits.robIdx) && io.brqRedirect.valid && vecExceptionFlag.valid) && allocated(i) && !committed(i)
```

Comment:
Overly long line. suggest using a conditional expression instead:
```suggestion
    needCancel(i) := allocated(i) && !committed(i) && Mux(
      vecExceptionFlag.valid,
      ...,
      ...
    )
```